config/routes.rb

```diff
@@ -1,10 +1,13 @@
 Rails.application.routes.draw do
   root to: 'repositories#index'
 
-  get 'repo/*repo/tag/*tag', to: 'tags#show',         as: :tag       , constraints: { repo: /.+/, tag: /[^\/]+/ }
-  get 'repo/*repo',          to: 'repositories#show', as: :repository, constraints: { repo: /.+/ }
+  get 'repo/*repo/tag/*tag',    to: 'tags#show',         as: :tag,               constraints: {repo: /.+/, tag: /[^\/]+/}
+  get 'repo/*repo',             to: 'repositories#show', as: :repository,        constraints: {repo: /.+/}
 
-  delete 'repo/*repo/tag/*tag', to: 'tags#destroy', constraints: { repo: /.+/, tag: /[^\/]+/ }
+  get '*repo:*tag',             to: 'tags#show',         as: :direct_tag,        constraints: {repo: /.+/, tag: /[^\/]+/}
```

Comment:
I think we don't need a name for this additional route since it's not used in any of the views.

```suggestion
  get '*repo:*tag',             to: 'tags#show',                                 constraints: {repo: /.+/, tag: /[^\/]+/}
```